aws/resource_aws_iot_thing_type.go

```diff
@@ -0,0 +1,186 @@
+package aws
+
+import (
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iot"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// https://docs.aws.amazon.com/iot/latest/apireference/API_CreateThingType.html
+func resourceAwsIotThingType() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsIotThingTypeCreate,
+		Read:   resourceAwsIotThingTypeRead,
+		Update: resourceAwsIotThingTypeUpdate,
+		Delete: resourceAwsIotThingTypeDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				d.Set("name", d.Id())
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIotThingTypeName,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateIotThingTypeDescription,
+			},
+			"searchable_attributes": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				MaxItems: 3,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateIotThingTypeSearchableAttribute,
+				},
+			},
```

Comment:
Is there any particular reason we should be drifting away from the API here? i.e. not nest `description` and `searchable_attributes` under `properties`?

Obviously we don't know how is the API going to evolve. It's a guessing game at this point, but it has proven to me in the past that it's better to stick to the API. 😉